From cd83b457d53e8d220e825d582e29033553efea2a Mon Sep 17 00:00:00 2001
From: Anand Ashok Dumbre <anand.ashok.dumbre@xilinx.com>
Date: Mon, 1 Mar 2021 22:43:41 -0800
Subject: [PATCH 1744/1851] iio: adc: versal-sysmon: Additional documentation
 for production silicon support.

commit 49bdb4e70d2001f3a5bf0bf5e990cca4a85210f4 from
https://github.com/Xilinx/linux-xlnx.git

production silicon uses one temperature value for alarms and TEMP_MIN
crossing the lower threshold no longer generates alarms.

Alarm generation condition for production silicon:
	temp > upper threshold
	temp < lower threshold

for ES1 silicon, temp channel referes to TEMP_MAX value.
Alarm generation condition for ES1 silicon:
	temp > upper threshold
	TEMP_MIN < lower threshold.

Signed-off-by: Anand Ashok Dumbre <anand.ashok.dumbre@xilinx.com>
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/iio/adc/versal-sysmon.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/iio/adc/versal-sysmon.c b/drivers/iio/adc/versal-sysmon.c
index e67f57e428fc..661518593b94 100644
--- a/drivers/iio/adc/versal-sysmon.c
+++ b/drivers/iio/adc/versal-sysmon.c
@@ -24,8 +24,18 @@
 #include <linux/of_address.h>
 
 /* Channel IDs for Temp Channels */
+/* TEMP_MAX gives the current temperature for Production
+ * silicon.
+ * TEMP_MAX gives the current maximum temperature for ES1
+ * silicon.
+ */
 #define TEMP_MAX	160
+
+/* TEMP_MIN is not applicable for Production silicon.
+ * TEMP_MIN gives the current minimum temperature for ES1 silicon.
+ */
 #define TEMP_MIN	161
+
 #define TEMP_MAX_MAX	162
 #define TEMP_MIN_MIN	163
 #define TEMP_EVENT	164
@@ -179,7 +189,7 @@ static const struct iio_event_spec sysmon_supply_events[] = {
 
 /* Temperature channel attributes */
 static const struct iio_chan_spec temp_channels[] = {
-	SYSMON_CHAN_TEMP(TEMP_MAX, "max"),
+	SYSMON_CHAN_TEMP(TEMP_MAX, "temp"),
 	SYSMON_CHAN_TEMP(TEMP_MIN, "min"),
 	SYSMON_CHAN_TEMP(TEMP_MAX_MAX, "max_max"),
 	SYSMON_CHAN_TEMP(TEMP_MIN_MIN, "min_min"),
-- 
2.31.1

